src/render/graph.rs

```diff
@@ -45,6 +47,10 @@ impl Node {
             .process(&self.inputs[..], &mut self.outputs[..], params, scope)
     }
 
+    fn event_emitter(&mut self) -> Option<&mut dyn EventEmitter>{
```

Comment:
I think it would be better to put the event emitter in the `RenderScope` object so you can directly call it from the `process` call of the AudioProcessor. We of course hide it from the public API

Reply:
Yeah that's a possibility indeed, I was in the model of

- thread.render()
- thread.check_event();
- thread.handle_incoming_messages();

but maybe I'm just overcomplicate things here